scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/DatasetService.java

```diff
@@ -0,0 +1,368 @@
+        String name;
+        if (dataset.isPresent()) {
+            username = dataset.get().getUsername();
+            name = dataset.get().getName();
+            Path path = createPath(username, name);
+            File directory = path.toFile();
+            try {
+                success = deleteDirectory(directory);
+            } catch (SecurityException e) {
+                throw new DatasetException("A SecurityException occurred in deleteDatasetDirectories", e);
+            }
+        }
+        return success;
+    }
+
+    /**
+     * Helper method to recursively delete directories.
+     *
+     * @param directoryToBeDeleted The directory / File to be deleted.
+     * @return Returns true if the delete was successful, false if it failed.
+     * @throws SecurityException if directory cannot be deleted.
+     */
+    private boolean deleteDirectory(File directoryToBeDeleted) throws DatasetException {
+        boolean returnValue;
+        File[] allContents;
+        try {
+            allContents = directoryToBeDeleted.listFiles();
+        } catch (SecurityException e) {
+            throw new DatasetException("A SecurityException occurred in deleteDirectory when generating file list", e);
+        }
+        if (allContents != null) {
+            for (File file : allContents) {
+                deleteDirectory(file);
+            }
+        }
+        try {
+            returnValue = directoryToBeDeleted.delete();
+        } catch (SecurityException e) {
+            throw new DatasetException("A SecurityException occurred in deleteDirectory when deleting a directory", e);
+        }
+        return returnValue;
+    }
+
+    /**
+     * A helper method to create a Path given a username and dataset name.
+     *
+     * @param username    The username of the owner of the dataset.
+     * @param datasetName The name of the dataset.
+     * @return Returns a Path.
+     */
+    private Path createPath(String username, String datasetName) {
+        Path path;
+        if (datasetName != null) {
+            try {
+                path = Paths.get(this.rootLocationString, username, datasetName);
+                String pathName = StringUtils.cleanPath(Objects.requireNonNull(path).toString());
+                path = Paths.get(pathName);
+                return path;
+            } catch (InvalidPathException e) {
+                throw new DatasetException("An InvalidPathException occurred in createPath with null datasetName", e);
+            }
+        } else {
+            try {
+                path = Paths.get(this.rootLocationString, username);
+                String pathName = StringUtils.cleanPath(Objects.requireNonNull(path).toString());
+                path = Paths.get(pathName);
+            } catch (InvalidPathException e) {
+                throw new DatasetException("An InvalidPathException occurred in createPath with datasetName present", e);
+            }
+            return path;
+        }
+    }
+}
```

Comment:
You have two classes that do file system operations, they should probably merged into one class that is solely responsible for file system operations and your two service classes use that file system class.